src/test/java/com/timgroup/statsd/UnixSocketTest.java

```diff
@@ -134,16 +155,28 @@ public void resist_dsd_restart() throws Exception {
         while(lastException.getMessage() == null) {
             Thread.sleep(10);
         }
-        assertThat(lastException.getMessage(), containsString("No such file or directory"));
+        // Depending on the state of the client at that point we might get different messages.
+        assertThat(lastException.getMessage(), anyOf(containsString("No such file or directory"), containsString("Connection refused")));
```

Comment:
When would we get "Connection refused" when connecting to a non-existent socket?

It feels like there is a race condition here somewhere, where stream client gets out of sync with the test and why the second attempt down below is necessary.

Reply:
I honestly don't remember has this was written quite some time ago now, and wasn't able to reproduce the issue. Removed